### PR TITLE
refactor: rebrand Modal component

### PR DESCRIPTION
**Pull request by chaitanyadeorukhkar**
Old vs New (Ignore radio buttons, they're not migrated yet so removed them)
Screenshot 2023-12-26 at 11 01 54 AM

### CODE REVIEW

File: packages/blade/src/components/BaseHeaderFooter/BaseHeader.tsx [hunk 1]
```diff
@@ -229,7 +229,7 @@ const _BaseHeader = ({
                 flexDirection="row"
               >
                 {title ? (
-                  <Text size="large" weight="semibold" type="normal">
+                  <Text size="large" weight="semibold" color="surface.text.gray.normal">
```
Comment (kamleshchandnani) [inline packages/blade/src/components/BaseHeaderFooter/BaseHeader.tsx:232]: I hope you have verified the `weight` with the new mapping of typography?

File: packages/blade/src/components/Modal/Modal.web.tsx [hunk 2]
```diff
@@ -132,6 +132,7 @@ const Modal = ({
     }
     return null;
   });
+  // return <div>Hello</div>;
```
Comment (kamleshchandnani) [inline packages/blade/src/components/Modal/Modal.web.tsx:135]: remove?

File: packages/blade/src/components/BaseHeaderFooter/BaseFooter.tsx [hunk 3]
```diff
@@ -19,7 +19,7 @@ const _BaseFooter = ({
 }: BaseFooterProps): React.ReactElement => {
   return (
     <>
-      {showDivider && <Divider />}
+      {showDivider && <Divider variant="muted" />}
```
Comment (kamleshchandnani) [inline packages/blade/src/components/BaseHeaderFooter/BaseFooter.tsx:22]: remove this we shall make muted our default variant.

File: packages/blade/src/components/BaseHeaderFooter/BaseHeader.tsx (hunk 1 above)
Comment (chaitanyadeorukhkar) [reply packages/blade/src/components/BaseHeaderFooter/BaseHeader.tsx:232]: Yup. Its semibold (600) on figma as well

**Review by kamleshchandnani — APPROVED** [review]

**Pull request merged**